community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/commands/Predicate.scala

```diff
@@ -251,8 +251,8 @@ case class PropertyExists(identifier: Expression, propertyKey: KeyToken) extends
   override def localEffects = Effects.READS_ENTITIES
 }
 
-case class LiteralRegularExpression(a: Expression, regex: Literal) extends Predicate {
-  lazy val pattern = regex.v.asInstanceOf[String].r.pattern
+case class LiteralRegularExpression(a: Expression, regex: Literal)(implicit converter: String => String = ((a) => a)) extends Predicate {
```

Comment:
@davidegrohmann I didn't change anything here apart from the converter function? However I am not crazy about the cast and I am not sure why `pattern` is lazy so refactoring might be a good idea.